### DISCUSSION TITLE
AMD RX 570 - 580 COMMANDLINE_ARGS=

### DISCUSSION THREAD

**Discussion by SunGreen777** [lshqqytiger/stable-diffusion-webui-directml#223] [General]
Hello rx570-580 owners
There were many changes, knowledgeable people please tell me what is in your config in COMMANDLINE_ARGS=
On version 1.51

**Comment by Siralob** [comment]
Hello. Did you check [ths thread](https://github.com/lshqqytiger/stable-diffusion-webui-directml/discussions/84)? It's worth to read whole discussions. It helps me a lot.
It will suit RX470, 480 too. (gfx803 a.k.a polaris)
```
set PYTORCH_CUDA_ALLOC_CONF=garbage_collection_threshold:0.8,max_split_size_mb:512
set COMMANDLINE_ARGS= --medvram --no-half --precision full --disable-nan-check
```
`--precision full`
also, `--no-half-vae` can reduce plane image randomly appears.

On the v1.5.1

**Reply by fishonainternet** [reply]
Can you do hires fix with this?

**Reply by Siralob** [reply]
In short answer, I can' use it, because RX580 8gb can't handle it. I got oom message when I did hires-fix with `--no-half`. 
But you can do hires-fix with different arguments. 

1. All you need is delete arguments `--no-half` , `--no-half-vae ` , `--precision full` , then launch `webui-user.bat`. I only used `--medvram` and `--disable-nan-check` in it.
I tested that it can generate 1024px images, which is twice as large as 512px. But it takes a long time to generate just single image and I can't use anything related to in-painting.

2. Using `--lowvram` I haven't tested on 1.5.1 yet (sry, I'm lazy :]]) but it will work, though. Cons: I experienced it was very slow to generate any resolution. It will be at least twice as slow as `--medvram` with my card. And the power consumption is also longer because the GPU compute hard for a long time. So, I did not try any more because of that.
update: I can't do it due to oom. 

You can use Ultimate SD upscaler instead of hires.fix. some of instruction video on youtube.

**Comment by VeteranXT** [comment]
I tried to use new SD XL and unable to create anyhing.

**Comment by LucasJackrabbit** [comment]
I have this:

`COMMANDLINE_ARGS= --always-batch-cond-uncond --medvram --precision autocast --disable-nan-check`

Everything else I add makes things usually worse in some way.
I don't use Inpaint, but if I need to I can switch to full precision by changing argument anytime.

**Reply by PVZ82** [reply]
I have a ryzen 4700u with 32Gb dedicated to the IGPU. I can use SDXL and generate 8 images of 1024x1024 in succession. Then the VRAM is full, and I have to restart the code.  I cannot run with -medvram... have to use -lowvram... with -medvram one image will not generate even, all 32GB vram is used and then the program stops. The current code has a lot of memory leakage.

**Comment by yun15634** [comment]
I don't know if it will be helpful, but
In my case, I needed a bigger swap file when using SDXL on RX570 8GB.
After that, I was able to run 1600x1024.
SD 1.5 can also be executed. (Hires.fix 512x512 to 768x768)

# spec
OS = Windows11 64bit
RAM = RAM32GB (Swap File 32-40GB)
GPU = Geforce GT635 + Radeon RX570 8GB (1000/1650MHz)
WEB-UI = lshqqytiger / Stable Diffusion web UI with DirectML
WEB-UI Optimize = sdp - scaled dot product

# WebUI_user.bat
set PYTORCH_CUDA_ALLOC_CONF=garbage_collection_threshold:0.6,max_split_size_mb:64
set COMMANDLINE_ARGS= --backend directml --lowvram --precision autocast --disable-nan-check
set CUDA_VISIBLE_DEVICES=1 *for Multi GPU Selection
*If inpaint cannot be used, add --no-half for COMMANDLINE_ARGS  (VRAM usage +25%)

# SDXL 1.0 / 1600x1024 (50steps)
>Steps: 50, Sampler: DPM++ 3M SDE Karras, CFG scale: 7.5, Seed: 536594023, Size: 1600x1024, Model hash: 06e890b5be, Model: hassakuSdxlAlphav_alphaV01, Clip skip: 2, Token merging ratio: 0.4, RNG: CPU, NGMS : 4, Eta: 0.67, Pad conds: True, VAE Decoder: TAESD, Version: 1.6.0
>Time taken: 17 min. 56.6 sec.
>A: 6.98 GB, R: 8.00 GB, Sys: 8.0/8 GB (100.0%)
![SD2 0_Tune_Low4](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/26879151/e1587af0-eb1c-4d19-8e49-1e88fc7d86bf)
https://i.imgur.com/qqYmTR8.png

# SD1.5 / 512x512 to 768x768 (Hires.fix 50/20steps)
*LowVRAM 3GB (4Minutes), MedVRAM 7GB (2minutes)
*Hires.fix 512x512 to 1280x1280  Works on LowVRAM
>Steps: 50, Sampler: DPM++ 2M Karras, CFG scale: 7.5, Seed: 536594030, Size: 512x512, Model hash: ef53c3c106, Model: BloodOrangeMixHalf_fubuki-v1e13_AOM3A1_OrangeMixVAE_v1, Denoising strength: 0.7, Clip skip: 2, Token merging ratio: 0.4, Token merging ratio hr: 0.5, RNG: CPU, NGMS: 4, Hires upscale: 1.5, Hires steps: 20, Hires upscaler: Lanczos, TI hashes: "EasyNegative: c74b4e810b03", Pad conds: True, VAE Encoder: TAESD, Refiner: AOM3A1_orangemixs [f303d10812], Refiner switch at: 0.8, Version: 1.6.0
>Time taken: 3 min. 55.1 sec.
>A: 3.15 GB, R: 8.00 GB, Sys: 8.0/8 GB (100.0%)
![SD1 5_Tune_Low1](https://github.com/lshqqytiger/stable-diffusion-webui-directml/assets/26879151/56ab3a37-d8bf-4ca3-a44a-1c76db18c48c)
https://i.imgur.com/pcdQQmD.png


# We recommend preparing 4 batch files. for RX570 8GB

High Resolution / No Support FP32 Plugin (Inpaint etc) / Hires.fix 512x512 to 1280x1280 Works
--backend directml --lowvram --precision autocast --disable-nan-check

High Resolution / Support FP32 Plugin (Inpaint etc) / VRAM usage +25% (Use this batch file only when using inpaint and some plugins)
--backend directml --lowvram --precision autocast --disable-nan-check --no-half

Mid Resolution / No Support FP32 Plugin (Inpaint etc)
--backend directml --medvram --precision autocast --disable-nan-check

Mid Resolution / Support FP32 Plugin (Inpaint etc) / VRAM usage +25% (Use this batch file only when using inpaint and some plugins)
--backend directml --medvram --precision autocast --disable-nan-check --no-half

**Reply by songib** [reply]
Amazing breakdown, i had 4 batch files aswell for hiresfix and inpaint, but mostly using inpaint mode. xd
i use `--precision autocast` before but i cant inpaint, maybe because i had `--no-half` in the setting before (i dont rememeber). ill try this out for sure.

using this atm `--backend directml --medvram --no-half --disable-safe-unpickle --skip-torch-cuda-test --always-batch-cond-uncond --disable-nan-check --api --autolaunch --theme dark --use-cpu CLIP BLIP interrogate`

ill try to add `--precision autocast` in my next run.

edit: ill try token merging aswell, but i always conscious about this since it can't reproduce the image accurately with the same seed or something. idk
Maybe my info mixed up.

**Comment by Bleach665** [comment]
In the GUI `Optimization` / `DirectML memory stats provider` set value to `atiadlxx (AMD only)`. After restart `stable-diffusion-webui-amdgpu`. This increased performance by ~40% for me. RX 580 2048SP.
My args: `COMMANDLINE_ARGS= --use-directml --lowvram --theme dark --precision autocast --skip-version-check`
